addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/KodiAudioSink.java

```diff
@@ -69,15 +70,18 @@ public void process(AudioStream audioStream) throws UnsupportedAudioFormatExcept
             // it is an external URL, the speaker can access it itself and play it.
             URLAudioStream urlAudioStream = (URLAudioStream) audioStream;
             url = urlAudioStream.getURL();
-        } else {
+        } else if (audioStream instanceof FixedLengthAudioStream) {
+            FixedLengthAudioStream fixedLengthAudioStream = (FixedLengthAudioStream) audioStream;
             if (callbackUrl != null) {
-                // we serve it on our own HTTP server
-                String relativeUrl = audioHTTPServer.serve(audioStream);
+                // we serve it on our own HTTP server for 30 seconds as Kodi requests the stream several times
+                String relativeUrl = audioHTTPServer.serve(fixedLengthAudioStream, 30);
                 url = callbackUrl + relativeUrl;
             } else {
                 logger.warn("We do not have any callback url, so kodi cannot play the audio stream!");
                 return;
             }
+        } else {
+            throw new UnsupportedAudioFormatException("Kodi can only handle FixedLengthAudioStreams.", null);
```

Comment:
No, sorry. I think we have to address @kaikreuzer here.

I understand what you are meaning. Maybe we should introduce a new exception class `UnsupportedAudioStreamException` for this case.